### DISCUSSION TITLE
Cannot rename abandoned PR to trigger replacement PR? (AzureDevOps Server)

### DISCUSSION THREAD

**Discussion by japj** [renovatebot/renovate#10456] [Request Help]
Hello,

I am using AzureDevOps Server (2020) and it seems I cannot rename an abandoned PullRequest.
For regular PRs you can simply 'click' in the title and adjust/save (e.g. rebase! works fine for existing PRs)
But for abandoned PRs this does not seem to work.

Is someone else able to rename abandoned PullRequests in AzureDevOps (Server)?

**Comment by rarkins** [comment]
@japj what's the reason for needing to rename - is it for the onboarding PR or a normal upgrade PR?

@JamieMagee do you know?

**Reply by japj** [reply]
I want to retrigger a normal upgrade PR as per the instruction

Renovate left a comment on the abandoned PR:

```txt
Renovate Ignore Notification
As this PR has been closed unmerged, Renovate will ignore this upgrade and you will not receive PRs for any future 7.x releases. However, if you upgrade to 7.x manually then Renovate will then reenable updates for minor and patch updates automatically.

If this PR was closed by mistake or you changed your mind, you can simply rename this PR and you will soon get a fresh replacement PR opened.
```

**Comment by froque** [comment]
This also affects bitbucket. 
I declined a pull request by mistake. Renovate commented: 
> If this PR was closed by mistake or you changed your mind, you can simply rename this PR and you will soon get a fresh replacement PR opened.

But renaming declined pull requests is not allowed in bitbucket (https://jira.atlassian.com/browse/BCLOUD-15190)

**Comment by rarkins** [comment]
OK so for platform which don't support renaming of closed PRs, we need an alternative method to "invalidate" them to trigger new ones. This could include labels (ideal, but maybe not supported by BB?), PR comments (non-ideal, because it increases the API calls we need to make substantially), or perhaps any other technique someone can think of.

**Comment by horihel** [comment]
For anyone who comes around this - here's my procedure/workaround to re-onboard in AzDo after renovate has identified the repo as disabled (and deleted the onboarding branch)

- create fresh branch "renovate/configure"
- now you can reactivate your configuration PR
- now you can rename the PR
- abandon the renamed onboarding PR
- delete the branch "renovate/configure"

renovate can now create a fresh onboarding procedure on your repo the next time it runs

**Reply by viceice** [reply]
@HonkingGoose Can we add this to our FAQ or the the azure devops docs?

**Reply by HonkingGoose** [reply]
> @HonkingGoose Can we add this to our FAQ or the the azure devops docs?

I'm hesitant to add this to our docs, as it's only a workaround, not a proper fix. Possible solutions:

1. Fix the problem in the code/behavior of the program
2. Else, change the "refresh pr instructions" so they're platform specific, and can explain the workarounds".
3. As a last resort: add workaround instructions to the docs. 😉 

---

**Solution 2:** 

Example of a possible platform specific ignore message:

> ### Renovate Ignore Notification
> 
> As this PR has been closed unmerged, Renovate will ignore this upgrade and you will not receive PRs for *any* future 16.x releases. However, if you upgrade to 16.x manually then Renovate will reenable minor and patch updates automatically.
> 
> If this PR was closed by mistake or you changed your mind, you can simply rename this PR and you will soon get a fresh replacement PR opened.
> 
> To re-open the PR on Azure DevOps do this instead:
> 
> 1. Step one.
> 2. Step two
> 3. Step three.

**Reply by viceice** [reply]
@HonkingGoose This is not a renovate issue, it's just a limitation on azure devops platform. You can't change a PR, if the branch is missing. So this is a good workaround we should communicate to our users.

**Reply by rarkins** [reply]
We could hack our platform massage markdown function to change the description for AZDO

**Reply by horihel** [reply]
btw another addition to the workaround:
This only works for PRs that have been abandoned. Completed PRs cannot be reactivated - which means (AFAIK) those are totally immutable.

**Reply by GK-Matias** [reply]
Is there a solution to re-onboarding if the original onboarding PR has been completed or would that require recreating the repo?

**Reply by horihel** [reply]
AFAIK your only solution is manually adding/editing the config.

**Reply by AlexDanDuna** [reply]
@horihel Thank you for sharing this workaround. However, I must say I was unable to apply it because I did not know what the branch 'renovate/configure' was referring to. It would be nice if you changed the branch name to something like "name of the branch to be merged in the PR'. Wish I figured this out on my own a couple of hours ago when I read your post before finding this workaround somewhere else :). Great idea nonetheless and thank you for sharing it once again.

**Comment by cjakobsen** [comment]
Came across this discussion and just want to describe what I did to resolve this on Azure DevOps.

I had abandoned a PR that covered version 37.x so I had effectively skipped out of all new versions until next major version, not what I wanted. I was able to reactive the orginal PR and Renovate started giving me the new minor versions as I wanted. What I did was this:

1. Reactivate the PR according to this [Reactivate a PR](https://learn.microsoft.com/en-us/azure/devops/repos/git/complete-pull-requests?view=azure-devops&tabs=browser#abandon-or-reactivate-a-pull-request).
2. If the Reactivate button is disabled the source branch is probably deleted and has to be restored before the PR can be reactivated, this can be done using this [Restore deleted branch](https://learn.microsoft.com/en-us/azure/devops/repos/git/restore-deleted-branch?view=azure-devops).

**Reply by gioce90** [reply]
This is a good approach. 
I'm asking if deleted branches will be deleted forever after some time.

In that case there should be another approach to trigger again the PR.

**Comment by gioce90** [comment]
The solutions suggested from @horihel and @cjakobsen  are good, but maybe a more direct approach it would be better.

In Azure is possible to add labels also on closed PR. Maybe a "renovate-reactive" label can be a solution?
